src/components/proto/menu.svelte

```diff
@@ -1,128 +1,200 @@
 <script lang="ts">
-  import { onMount } from 'svelte';
-
-  onMount(() => {
-    const openButton = document.getElementById('open-button');
-    const backButton = document.getElementById('back-button');
-    const menu = document.getElementById('menu');
-    if (openButton != null) {
-      openButton.onclick = () => {
-        if (menu != null) menu.style.display = '';
-      };
+  let menuOpen = false;
+  let menuStack = ['menu-contents-0'];
+  let menuStackLength = menuStack.length;
+  let visibleClass = 'visible';
+
+  const openMenu = () => {
+    menuOpen = true;
+  };
+
+  const closeMenu = () => {
+    menuOpen = false;
+  };
+
+  const push = (id: string) => {
+    const lastElementId = menuStack[menuStack.length - 1];
+    // check if element id is already on the stack
+    if (lastElementId == id) {
+      console.warn(`${id} is already at end of stack`);
+      return;
+    }
+    const elem = document.getElementById(id);
+    if (elem == null) {
+      console.error(`Element with id ${id} does not exist! Ignoring push.`);
+      return;
+    }
+    const lastElem = document.getElementById(lastElementId);
+    if (lastElem != null) {
+      lastElem.classList.remove(visibleClass);
+      lastElem.classList.add('pushed'); // moves the element to the left
     }
+    elem.classList.add(visibleClass);
+    menuStack.push(id);
+    menuStackLength = menuStack.length;
+  };
 
-    if (backButton != null) {
-      backButton.onclick = () => {
-        if (menu != null) menu.style.display = 'none';
-      };
+  const pop = () => {
```

Comment:
Should implement a feature to `pop` to a specific menu if necessary, such as the initial menu, or another submenu. 

Also, should menu sub-navigation also be done with keyboard hotkeys? Or perhaps allow routing to different submenus via a link? Perhaps that can be done at a later date.